charts/kuberpult/templates/manifest-repo-export-service.yaml

```diff
@@ -0,0 +1,213 @@
+# This file is part of kuberpult.
+
+# Kuberpult is free software: you can redistribute it and/or modify
+# it under the terms of the Expat(MIT) License as published by
+# the Free Software Foundation.
+
+# Kuberpult is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# MIT License for more details.
+
+# You should have received a copy of the MIT License
+# along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+# Copyright freiheit.com
+
+{{- if not (or (eq .Values.cd.db.dbOption "cloudsql") (eq .Values.cd.db.dbOption "sqlite") (eq .Values.cd.db.dbOption "NO_DB")) }}
+{{ fail ".Values.cd.db.dbOption does not contain a valid value (NO_DB, sqlite, cloudsql)."}}
+{{ end -}}
+
+# the export service is only enabled, if the DB is enabled
+{{- if or (eq .Values.cd.db.dbOption "cloudsql") (eq .Values.cd.db.dbOption "sqlite") }}
+
+{{- if eq .Values.cd.db.writeEslTableOnly false }}
+
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kuberpult-manifest-repo-export-service
+  labels:
+    app: kuberpult-manifest-repo-export-service
+{{- if .Values.datadogTracing.enabled }}
+    tags.datadoghq.com/service: kuberpult-manifest-repo-export-service
+    tags.datadoghq.com/version: {{ $.Chart.AppVersion }}
+    tags.datadoghq.com/env: {{ .Values.datadogTracing.environment }}
+{{- end }}
+spec:
+  # This service works as an export. There should only be 1 instance of it at all times.
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kuberpult-manifest-repo-export-service
+  template:
+    metadata:
+      labels:
+        app: kuberpult-manifest-repo-export-service
+{{- if .Values.datadogTracing.enabled }}
+        tags.datadoghq.com/env: {{ .Values.datadogTracing.environment }}
+        tags.datadoghq.com/service: kuberpult-manifest-repo-export-service
+        tags.datadoghq.com/version: {{ $.Chart.AppVersion }}
+      annotations:
+        apm.datadoghq.com/env: '{"DD_SERVICE":"kuberpult-manifest-repo-export-service","DD_ENV":"{{ .Values.datadogTracing.environment }}","DD_VERSION":"{{ $.Chart.AppVersion }}"}'
+{{- end }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+{{- if eq .Values.cd.db.dbOption "cloudsql" }}
+      serviceAccountName: {{ .Values.cd.db.k8sName }}
+{{- end }}
+      containers:
+{{- if eq .Values.cd.db.dbOption "cloudsql" }}
```

Comment:
Is this required for the service?
Also there are a lot of `cd` references in this file, is this correct?

Reply:
It's not technically required right now, but it will be in the next PR.
I do think we should rename `cd.db` to `cd`, but I would prefer to do this later.